cpp/src/parquet/encoding.cc

```diff
@@ -466,6 +465,15 @@ struct DictEncoderTraits<FLBAType> {
 // Initially 1024 elements
 static constexpr int32_t kInitialHashTableSize = 1 << 10;
 
+int RlePreserveBufferSize(int num_values, int bit_width) {
+  // Note: because of the way RleEncoder::CheckBufferFull() is called, we have to
+  // reserve
+  // an extra "RleEncoder::MinBufferSize" bytes. These extra bytes won't be used
+  // but not reserving them would cause the encoder to fail.
+  return 1 + ::arrow::util::RleEncoder::MaxBufferSize(bit_width, num_values) +
+         ::arrow::util::RleEncoder::MinBufferSize(bit_width);
```

Comment:
You could avoid this extraction by creating a local variable called `num_values`.

Note that the caller returns `int64_t`, so you can also return `int64_t` and avoid a potential `int` overflow when summing `a + int + int`.

Reply:
int64_t seems ok, but previously, the code using `RlePreserveBufferSize` never do a overflow checking. So I use int here.

What do you mean by creating local variable?

Reply:
I supposes you extracted a new function here for clarity, but you call in multiple places now?

Reply:
If the extract was just for the sake of clarify, creating a variable would be preferable. You can mark this comment thread as resolved. :)